b2share/modules/communities/cli.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of EUDAT B2Share.
+# Copyright (C) 2016 University of Tuebingen, CERN.
+# Copyright (C) 2015 University of Tuebingen.
+#
+# B2Share is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# B2Share is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with B2Share; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""B2Share Communities module Command Line Interface."""
+
+from __future__ import absolute_import
+
+from os.path import isfile
+
+import click
+from flask_cli import with_appcontext
+
+from invenio_db import db
+
+from .api import Community, CommunityDoesNotExistError
+
+#from .errors import RootSchemaAlreadyExistsError
+#from .helpers import load_root_schemas
+
+@click.group()
+def communities():
+    """communities management commands."""
+
+
+@communities.command()
+@with_appcontext
+@click.option('-v', '--verbose', is_flag=True, default=False)
+@click.argument('name')
+@click.argument('description')
+@click.argument('logo')
+def create(verbose, name, description, logo):
+    """Creating a community in the database. Name can be 255 characters long. Description is a text of maximally 1024 characters enclosed in parentheses. The logo parameter should be a valid path to a logo file. """
+    if len(name)>255:
+        raise click.BadParameter("NAME parameter is longer than the 255 character maximum")
+    if len(description)>1024:
+        raise click.BadParameter("DESCRIPTION parameter is longer than the 1024 character maximum")
+    if not isfile(logo):
```

Comment:
I assumed someone would place them in the right folder, since we're talking about administrators doing this operation. An alternative would be to accept a full URL, download the image stored at that url, and place it in the right folder. This would be certainly a nice to have feature.